onnxruntime/core/framework/error_code_helper.h

```diff
@@ -8,3 +8,10 @@
 namespace onnxruntime {
 OrtStatus* ToOrtStatus(const onnxruntime::common::Status& st);
 };
+
+#define API_IMPL_BEGIN try {
+#define API_IMPL_END                                          \
+  }                                                           \
+  catch (std::exception & ex) {                               \
+    return OrtCreateStatus(ORT_RUNTIME_EXCEPTION, ex.what()); \
```

Comment:
Would be better if there is another branch for the NotImplemented exception.

Reply:
Fixed, I can see us adding more specific type handlers in the future.